axelor-account/src/main/java/com/axelor/apps/account/web/AnalyticDistributionLineController.java

```diff
@@ -117,4 +127,60 @@ public void setAnalyticAxisDomain(ActionRequest request, ActionResponse response
               .getAnalyticAxisDomain(analyticMoveLine, company));
     }
   }
+
+  public void setAnalyticAccountDomain(ActionRequest request, ActionResponse response) {
+    String domain = "null";
+    List<Long> analyticAccountIdList = new ArrayList<>();
+    AnalyticDistributionLine analyticDistributionLine =
+        request.getContext().asType(AnalyticDistributionLine.class);
+    Context parentContext = request.getContext().getParent();
+    Context grandParentContext = null;
+
+    if (parentContext != null) {
+      AnalyticDistributionTemplate analyticDistributionTemplate =
+          parentContext.asType(AnalyticDistributionTemplate.class);
+
+      if (analyticDistributionTemplate != null
+          && analyticDistributionTemplate.getCompany() != null) {
+        domain =
+            "(self.company is null OR self.company.id = "
+                + analyticDistributionTemplate.getCompany().getId()
+                + ") AND self.analyticAxis.id ";
+        if (analyticDistributionLine.getAnalyticAxis() != null) {
+          domain += "= " + analyticDistributionLine.getAnalyticAxis().getId();
+        } else {
+          domain +=
+              "in ("
+                  + Beans.get(AccountConfigRepository.class)
+                      .findByCompany(analyticDistributionTemplate.getCompany())
+                      .getAnalyticAxisByCompanyList().stream()
+                      .map(it -> it.getAnalyticAxis().getId().toString())
+                      .collect(Collectors.toList())
+                  + ")";
+        }
+
+        grandParentContext = request.getContext().getParent().getParent();
+        if (grandParentContext != null) {
```

Comment:
Check here if the grandParentContext class is Account.class

Reply:
Added